tfjs-core/src/ops/conv_util.ts

```diff
@@ -496,6 +495,10 @@ function get3DPadAndOutInfo(
   let outHeight: number;
   let outWidth: number;
 
+  if (pad === 'valid') {
+    pad = 0;
+  }
```

Comment:
If pad is 'valid', it should have the same result as `pad = 0`.

By the way, the roundingMode for this case is supposed to be 'truncate', instead of 'ceil', referring to [tensorflow](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/kernel_shape_util.cc#L38).